### PR TITLE
Add ReadyToRun environment variable run-time option

### LINKED ISSUE #16781: Document "COMPlus_ReadyToRun"
AFAIK the only documentation for the `COMPlus_ReadyToRun` configuration is here: https://docs.microsoft.com/en-us/visualstudio/debugger/jit-optimization-and-debugging?view=vs-2019#limitations-of-the-suppress-jit-optimization-option

We should consider adding it here too.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 70b4e642-6b61-128c-3eba-2f0ca8ff440c
* Version Independent ID: 4fe316ee-08aa-2989-af97-2bfc0f698f01
* Content: [Compilation config settings](https://docs.microsoft.com/en-us/dotnet/core/run-time-config/compilation)
* Content Source: [docs/core/run-time-config/compilation.md](https://github.com/dotnet/docs/blob/master/docs/core/run-time-config/compilation.md)
* Product: **dotnet-core**
* GitHub Login: @gewarren
* Microsoft Alias: **gewarren**

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #16781

### CODE REVIEW

File: docs/core/run-time-config/compilation.md [hunk 1]
```diff
@@ -125,3 +125,13 @@ Project file:
 
 </Project>
 ```
+
+## ReadyToRun
+
+- Configures whether the .NET Core runtime uses pre-compiled images for framework code (and user assemblies, if pre-compiled images are available). Disabling this option forces the runtime to JIT-compile framework code.
```
Comment (gewarren) [inline docs/core/run-time-config/compilation.md]: Is the part about user assemblies true, or are the R2R images always used if they're available?

Comment (fadimounir) [inline docs/core/run-time-config/compilation.md]:
```suggestion
- Configures whether the .NET Core runtime uses pre-compiled code for images with available ReadyToRun data. Disabling this option forces the runtime to JIT-compile framework code.
```

Comment (fadimounir) [reply docs/core/run-time-config/compilation.md]: There is no distinction between user/framework assemblies for this variable

**Review by fadimounir — APPROVED** [review]
LGTM modulo the suggestion I had to not distinguish between user/framework assemblies

File: docs/core/run-time-config/compilation.md [hunk 2]
```diff
@@ -125,3 +125,13 @@ Project file:
...
+- For more information see [ReadyToRun](../whats-new/dotnet-core-3-0.md#readytorun-images).
+- Default: Enabled (`1`).
+
+| | Setting name | Values |
+| - | - | - |
+| **Environment variable** | `COMPlus_ReadyToRun` | `1` - enabled<br/>`0` - disabled |
```
Comment (fadimounir) [inline docs/core/run-time-config/compilation.md]: Can we clarify that `1` is the default?

Comment (gewarren) [reply docs/core/run-time-config/compilation.md]: It is clarified in the bulleted list. Is that okay? That's how we've done it for the other options too.

File: docs/core/run-time-config/compilation.md [hunk 3]
```diff
@@ -125,3 +125,13 @@ Project file:
 
 </Project>
 ```
+
+## ReadyToRun
+
+- Configures whether the .NET Core runtime uses pre-compiled code for images with available ReadyToRun data. Disabling this option forces the runtime to JIT-compile framework code.
+- For more information see [ReadyToRun](../whats-new/dotnet-core-3-0.md#readytorun-images).
```
Comment (gewarren) [inline docs/core/run-time-config/compilation.md]:
```suggestion
- For more information, see [ReadyToRun](../whats-new/dotnet-core-3-0.md#readytorun-images).
```

File: docs/core/run-time-config/compilation.md (hunk 2 above)
Comment (fadimounir) [reply docs/core/run-time-config/compilation.md]: Ah! didn't notice :). Thanks

**Review by fadimounir — APPROVED** [review]

**Review by Thraka — APPROVED** [review]

**Pull request merged**